### ISSUE TITLE
Debugger throws when debugging code which references a source map but the map file is missing.

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/vscode#13336]
- VSCode Version: 1.6.0-insider
- OS Version:

Steps to Reproduce:

1. install vscode-jsonrpc@^3.0.0-alpha.1
2. install @type/node
3. create tsconfig.json
```json
{
	"compilerOptions": {
		"module": "commonjs",
		"sourceMap": true
	}
}
```
4. create app.ts
```ts
import * as rpc from 'vscode-jsonrpc';

let logger: rpc.Logger  = {
	error: (message) => {},
	info: (message) => {},
	warn: (message) => {},
	log: (message) => {}
}

let connection = rpc.createMessageConnection(process.stdin, process.stdout, logger);
```
5. compile using tsc. ignore the compile errors. Not relevant for the problem
6. create launch.json and ENABLE source maps
7. set breakpoint in app.ts
8. debug
9. try to step into createMessageConnection

You get (see the exception on the call stack). I am pretty sure that before the debugger simply stepped into the JS code.

![capture](https://cloud.githubusercontent.com/assets/1931590/19184058/95424fe8-8c7b-11e6-883c-26a167d1ff55.png)

**Comment by isidorn** [comment]
Works for me on OSX - trying on windows...

![stepinjs](https://cloud.githubusercontent.com/assets/1926584/19184601/43afb078-8c7e-11e6-8597-63ec1657bd68.gif)

**Comment by isidorn** [comment]
@dbaeumer works for me on windows also. What version of node do you have? Do you also see this with our Visual studio stable?
@bpasero might be dirk's special setup - in that case will remove the important tag and push to october

![stepinjs](https://cloud.githubusercontent.com/assets/1926584/19184844/4747b306-8c7f-11e6-91f8-36a5c50bc76e.gif)

**Comment by dbaeumer** [comment]
I have node 6.5.0.

@bpasero I am pretty sure that this has nothing to do with the subst setup. It looks for a file on disk that doesn't exist and throws. I tested it with opening the same workspace from C:\ and it throws in the same way.

**Comment by dbaeumer** [comment]
@isidorn you need to step into 'createMessageConnection'. In both screen casts you are stepping into process.stdin / stdout getters. You need to step out and step in again. May be more times. Sorry for not making this more explicit.

**Comment by isidorn** [comment]
@dbaeumer thanks for the clarification. I can repro on Win, not on Mac.
Does not repro with stable vscode - it is a regression, investigating...

**Comment by isidorn** [comment]
Ok pushed a fix which handles the error better and fixes the issue on windows.
Please code review @roblourens @weinand  since I do not have deep understanding of code,

**Comment by roblourens** [comment]
This will also show up if that function fails for any other reason, like if the sourcemap path is a url that doesn't exist, or invalid inlined base64. Maybe that's less likely but I can try to move the error handling upstream.

**Comment by roblourens** [comment]
I pushed a fix for this that catches anything outside of _loadSourceMap... not sure if I should put it in 1.6 or not?

**Comment by isidorn** [comment]
@roblourens for now I would go with the original fix since it is smaller and already verified. Unles you see some direct benefits of going with you proposed solution for the current release?
Thanks for looking into it!

**Comment by roblourens** [comment]
Nah, let's go with the original fix for 1.6 and keep the other for later.
